mgail.py

```diff
@@ -144,12 +193,15 @@ def policy_stop_condition(state_, t, cost, trans_err, env_term_sig):
         loop_outputs = tf.while_loop(policy_stop_condition, policy_loop, [state_0, 0., 0., 0., False])
         self.policy.train(objective=loop_outputs[2])
 
-    def al_loss(self, d):
-        logit_agent, logit_expert = tf.split(axis=1, num_or_size_splits=2, value=d)
+    def al_loss(self, d=None, log_p=None, log_q=None, log_pq=None):
+        if not self.use_irl:
+            logit_agent, logit_expert = tf.split(axis=1, num_or_size_splits=2, value=d)
+            labels = tf.concat([tf.zeros_like(logit_agent), tf.ones_like(logit_expert)], 1)
+            d_cross_entropy = tf.nn.softmax_cross_entropy_with_logits(logits=d, labels=labels)
+        
+        else: # USING IRL
+            labels = tf.concat([1 - self.label, self.label], 1) # TODO: We need to fix this! 
+            d_cross_entropy = labels*(log_p-log_pq) + (1-labels)*(log_q-log_pq)
```

Comment:
I think we don't need the labels here. Also, when training the policy, we need to pass lprobs to the feed_dict as well.

I am not sure whether my solution is correct. Please discuss here. I think since we are trying maximizing the reward, so the loss should be the negative reward function. So according to the reward function in page 12,  it should look like this.
```suggestion
            d_cross_entropy = - (log_p - log_pq) + (log_q - log_pq)
```

Reply:
I am thinking the lprobs may not involve expert action_probs but just agent's in this case.

Reply:
I think you may be right about not needing the labels anymore, but not sure what you mean by lprobs may not involve expert action_probs?

I'm thinking the discriminator learns what good action_probs look like from the expert and then scores the generated policy action_probs according to that?

Reply:
I addressed changing the cross entropy loss here:

https://github.com/yipsang/mgail/pull/7/commits/c7402b9ba1a8471aa5dc7443a8046afd0bd29071

But not too sure about lprobs stuff :P let's continue discussion here